### ISSUE TITLE
FIND for tag! seems to be off by one

### ISSUE THREAD

**Issue by dsunanda** [red/red#4860]
**Describe the bug**
A find for a tag in a string returns one position too far.

**To reproduce**
```
find "aaa <bbb> ccc" <bbb>
== "bbb> ccc"

```

**Expected behavior**
```
find "aaa <bbb> ccc" <bbb>
== "<bbb> ccc"
```

**Platform version**

```
Red 0.6.4 for Windows built 19-Mar-2021/7:34:02  commit #7316b75
```

**Comment by greggirwin** [comment]
Manually forming the value is a workaround until this is fixed.
```
>> find "aaa <bbb> ccc" form <bbb>
== "<bbb> ccc"
```

**Comment by greggirwin** [comment]
Tags are a special case. `File!` matches R2's behavior. However we may want `ref!` to `form` with its sigil. 
```
>> find "aaa @bbb ccc" @bbb
== "bbb ccc"
>> form @bbb
== "bbb"
>> find "aaa @bbb ccc" form @bbb
== "bbb ccc"
>> find "aaa @bbb ccc" mold @bbb
== "@bbb ccc"

>> find "aaa %bbb ccc"  %bbb
== "bbb ccc"
>> find "aaa %bbb ccc" form %bbb
== "bbb ccc"
>> find "aaa %bbb ccc" mold %bbb
== "%bbb ccc"
```

**Comment by Oldes** [comment]
@dsunanda it's not off by one.. it just does string search... check this one:
```red
>> find "bbb aaa <bbb> ccc" <bbb>
== "bbb aaa <bbb> ccc"
```

**Comment by Oldes** [comment]
Btw.. it seems to be known issue also in R3 -> https://www.curecode.org/rebol3/ticket.rsp?id=1160

**Comment by dockimbel** [comment]
Fixed by commits for #4944.
